executor/README.md

```diff
@@ -0,0 +1,270 @@
+
+Example - setting and unsetting environment variables:
+
+`curl -X PATCH -d '{env: {VAR_A: "123", SOME_B: null}}' $EXECUTOR_ENDPOINT`
+
+Potential responses:
+
+| Code | HTTP Meaning        | Possible reason                          |
+| ---- | ------------------- | ---------------------------------------- |
+| 204  | No Content          | The request was succesfully processed. |
+| 400  | Malformed           | The request body contains an unknown key, or a key's value is the wrong type. |
+| 503  | Service Unavailable | The request was valid, and received, but there was an (ostensibly temporary) failure with mesos itself. |
+
+#### `GET /task/:taskid`
+
+For successful requests, the response body could contain:
+
+| key | type | description |
+|-----|------|-------------|
+| `exit_code` | int | The exit code of the primary command. |
+| `before_codes` | array of ints | The exit codes of the before commands. |
+| `after_codes` | array of ints | The exit codes of the after commands. |
+
+Potential responses:
+
+| Code | HTTP Meaning        | Possible reason                          |
+| ---- | ------------------- | ---------------------------------------- |
+| 200  | Okay          | The task id was valid, and the response body contains task data |
+| 404  | Not found           | The provided task id was unknown or invalid |
+| 503  | Service Unavailable | The request was valid, and received, but there was an (ostensibly temporary) failure with mesos itself. |
+
+### Example use cases
+
+Below are some examples of how to use the above features to solve some common problems.
+
+#### Conditionally preventing command execution
+
+In cases where only one copy of a certain job should be running, before and after commands could be used to acquire and release a lock. Assuming you've written some scripts for managing your lock, you could execute them by passing the additional data to the cook scheduler's job creation API:
+
+```json
+{
+	"command": "my-actual-command",
+	"before_commands": [{
+		"value": "./acquire-zk-lock.sh my-service"
+		"guard": true
+	}],
+	"after_commands":[{
+		"value": "./release-zk-lock.sh my-service"
+	}],
+	"uris": [{
+		"value": "http://my-file-server/release-zk-lock.sh",
+		"executable": true
+  	},{
+  		"value": "http://my-file-server/acquire-zk-lock.sh",
+  		"executable": true
+  	}]
+}
+```
+
+The `guard` flag is used to prevent execution of later commands if we can't acquire the lock (in this case, `acquire-zk-lock.sh` script will return a non-zero exit code). Note that the scripts for managing the lock could be completely reusable, and that the primary command did not need to change.
+
+#### Monitoring a running command
+
+Perhaps you'd like to tail a log created by your primary command. You could use an asynchronous before command to do so. It will be started before your primary command, and continue running in the background until the primary command (and all other synchronous commands) finish.
```

Comment:
That is the current behavior: SIGINT, wait ~10 seconds, SIGKILL.

https://github.com/mojotech/Cook/blob/7c0c8162a136f032d37fe01b8686ee0bb7b9edce/executor/cook/launcher.py#L52